src/bmi_cfe.c

```diff
@@ -812,16 +823,31 @@ int read_init_config_cfe(const char* config_file, cfe_state_struct* model)
         if (strcmp(param_key, "nash_storage_surface") == 0) {
 	  nash_storage_surface_string_val = strdup(param_value);
 	  is_nash_storage_surface_set     = TRUE;
+	  if ((param_units == NULL) || (strlen(param_units) < 1)) {
+#if CFE_DEBUG >= 1
+	    printf ("WARNING: [units] expected for '%s' in config file. [m]\n", param_key);
+#endif
+	  }
 	  continue;
         }
         if (strcmp(param_key, "Kinf_nash_surface") == 0) {
 	  model->nash_surface_params.K_infiltration = strtod(param_value, NULL);
 	  is_K_infiltration_nash_surface_set = TRUE;
+	  if ((param_units == NULL) || (strlen(param_units) < 1)) {
+#if CFE_DEBUG >= 1
+	    printf ("WARNING: [units] expected for '%s' in config file. [1/hour]\n", param_key);
+#endif
+	  }
 	  continue;
         }
 	if (strcmp(param_key, "retention_depth_nash_surface") == 0) {
 	  model->nash_surface_params.retention_depth = strtod(param_value, NULL);
 	  is_retention_depth_nash_surface_set        = TRUE;
+	  if ((param_units == NULL) || (strlen(param_units) < 1)) {
+#if CFE_DEBUG >= 1
+	    printf ("WARNING: [units] expected for '%s' in config file. [m]\n", param_key);
+#endif
```

Comment:
I've looked through the rest of the code and I don't see where the units are actually used for anything, so while printing a warning about missing units is OK, it is somewhat meaningless if the units value is largely ignored and nothing interesting is done with it.

Reply:
that's a valid point, we don't use the units anywhere for anything in the code, the only point I see here is to parse some information to let the user know that units are missing 😊